src/otx/cli/utils/help_formatter.py

```diff
@@ -0,0 +1,222 @@
+    """
+    if subcommand is None or subcommand in ("install"):
+        return []
+    from otx.engine import Engine
+
+    contents: list[Panel | Markdown] = [Markdown(INTRO_MARKDOWN)]
+    target_command = getattr(Engine, subcommand)
+    cli_usage = get_cli_usage_docstring(target_command)
+    if cli_usage is not None:
+        cli_usage += f"\n{VERBOSE_USAGE.format(subcommand=subcommand)}"
+        quick_start = Panel(Markdown(cli_usage), border_style="dim", title="Quick-Start", title_align="left")
+        contents.append(quick_start)
+    return contents
+
+
+class CustomHelpFormatter(RichHelpFormatter, DefaultHelpFormatter):
+    """A custom help formatter for OTX CLI.
+
+    This formatter extends the RichHelpFormatter and DefaultHelpFormatter classes to provide
+    a more detailed and customizable help output for OTX CLI.
+
+    Attributes:
+        verbose_level : int
+            The level of verbosity for the help output.
+        subcommand : str | None
+            The subcommand to render the guide for.
+
+    Methods:
+        add_usage(usage, actions, *args, **kwargs)
+            Add usage information to the help output.
+        add_argument(action)
+            Add an argument to the help output.
+        format_help()
+            Format the help output.
+    """
+
+    verbosity_dict = get_verbosity_subcommand()
+    verbose_level = verbosity_dict["verbosity"]
+    subcommand = verbosity_dict["subcommand"]
+
+    def add_usage(self, usage: str | None, actions: Iterable[argparse.Action], *args, **kwargs) -> None:
+        """Add usage information to the formatter.
+
+        Args:
+            usage (str | None): A string describing the usage of the program.
+            actions (Iterable[argparse.Action]): An list of argparse.Action objects.
+            *args (Any): Additional positional arguments to pass to the superclass method.
+            **kwargs (Any): Additional keyword arguments to pass to the superclass method.
+
+        Returns:
+            None
+        """
+        if self.subcommand in REQUIRED_ARGUMENTS:
+            if self.verbose_level == 0:
+                actions = []
+            elif self.verbose_level == 1:
+                actions = [action for action in actions if action.dest in REQUIRED_ARGUMENTS[self.subcommand]]
+
+        super().add_usage(usage, actions, *args, **kwargs)
+
+    def add_argument(self, action: argparse.Action) -> None:
+        """Add an argument to the help formatter.
+
+        If the verbose level is set to 0, the argument is not added.
+        If the verbose level is set to 1 and the argument is not in the non-skip list, the argument is not added.
+
+        Args:
+            action (argparse.Action): The action to add to the help formatter.
+        """
+        if self.subcommand in REQUIRED_ARGUMENTS:
+            if self.verbose_level == 0:
+                return
+            if self.verbose_level == 1 and action.dest not in REQUIRED_ARGUMENTS[self.subcommand]:
```

Comment:
Is this safe if an user give an arbitrary subcommand such as `trani` not `train`?

Reply:
Raises an error for disallowed subcommands, which are arbitrarily prefixed in argparse.
<img width="748" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/38045080/910818c5-ec9c-49be-8720-b4b42dc0accd">